.github/workflows/unpublish.yml

```diff
@@ -0,0 +1,39 @@
+name: ⛔ Deprecate
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to update'
+        required: true
+        type: choice
+        options:
+        - edge
+        - latest
+      version:
+        description: 'Version to deprecate'
+        required: true
+      tag_version:
+        description: 'Version to use instead'
+        required: true
+
+jobs:
+  deprecate:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Setup Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: '20.x'
+        registry-url: 'https://registry.npmjs.org'
+
+    - name: Deprecate version
+      run: npm deprecate cloudinary-video-player@${{ github.event.inputs.version }} "This version of cloudinary-video-player has been deprecated."
```

Comment:
I really think we rarely if ever want to deprecate a version. I am not sure if this is also how you think of it as well, but regarding this workflow we need to make sure that it is clear this is not about reverting or rolling back a version this is about deprecating a version. 

That being said if you want now to just make latest point a to a previous version (rollback), how do you do that?

Reply:
This workflow _is_ about rolling-back/unpublishing.
Unfortunately since the left-pad incident you can not unpublish a package if you have dependant packages.
Our best option, if we have a faulty version is to deprecate it and set the latest/edge tag to the previous valid version, and that's what this workflow is doing

Reply:
@tsi what is the left pad incident? 
Anyway i am bit afraid of this workflow, why can not we have something that changes where the latest tag points to? also deprecating usually happens in favor of a new version but here we are saying it happens in favor of an old version, not sure how this eventually will play out?

Reply:
https://qz.com/646467/how-one-programmer-broke-the-internet-by-deleting-a-tiny-piece-of-code 😄 

Note that this workflow does 2 things:
1. `npm deprecate`
2. `npm dist-tag` which is doing what you said - moving where the latest/edge tag points to
the `deprecate` part is just a "bonus" to let future users know that this version should not be used.